### PR TITLE
Tide: query Gerrit changes in parallel

### PR DESCRIPTION
**Pull request by chaodaiG**
Should greatly speed up tide and the turnaround time would be better

/cc @cjwagner @listx

### CODE REVIEW

File: prow/tide/gerrit.go [hunk 1]
```diff
@@ -132,23 +134,56 @@ func (p *GerritProvider) Query() (map[string]CodeReviewCommon, error) {
...
+	go func() {
+		for {
+			select {
+			case err := <-errChan:
+				combinedErrs = append(combinedErrs, err)
+				wg.Done()
+			case changes := <-resChan:
+				for _, pr := range changes.changes {
+					crc := CodeReviewCommonFromGerrit(&pr, changes.instance)
+					res[prKey(crc)] = *crc
```
Comment (listx) [inline prow/tide/gerrit.go:174]: Nit: Where does `prKey()` come from?

File: prow/tide/gerrit.go [hunk 2]
```diff
@@ -132,23 +134,56 @@ func (p *GerritProvider) Query() (map[string]CodeReviewCommon, error) {
...
+				}
+				wg.Done()
+			}
+		}
+	}()
+
+	wg.Wait()
+
+	// Let's not return error unless all queries failed.
```
Comment (listx) [inline prow/tide/gerrit.go:183]: Nit: Should we bother logging for the case where some (not all) queries failed?

**Review by listx — COMMENTED** [review]
Just nits.

/lgtm
/hold

File: prow/tide/gerrit.go (hunk 1 above)
Comment (chaodaiG) [reply prow/tide/gerrit.go:174]: it's a from tide.go

File: prow/tide/gerrit.go (hunk 2 above)
Comment (chaodaiG) [reply prow/tide/gerrit.go:183]: it's already logged above as warnings

**Pull request merged**